src/prefect/server/concurrency/lease_storage/memory.py

```diff
@@ -0,0 +1,60 @@
+from datetime import datetime, timedelta, timezone
+from uuid import UUID, uuid4
+
+from prefect.server.concurrency.lease_storage import (
+    ConcurrencyLeaseStorage as _ConcurrencyLeaseStorage,
+)
+from prefect.server.concurrency.lease_storage import (
+    ConcurrencyLimitLeaseMetadata,
+)
+from prefect.server.utilities.leasing import ResourceLease
+
+
+class ConcurrencyLeaseStorage(_ConcurrencyLeaseStorage):
```

Comment:
This is also named `ConcurrencyLeaseStorage` because I want to follow the pattern set in `prefect.server.utilites.messaging` where an alternative implementation can be used by pointing to a module that follows the expected interface.